### ISSUE TITLE
Type is inferred incorrectly for an infinite generator function

### ISSUE THREAD

**Issue by decorator-factory** [microsoft/pyright#2295]
Reproduced with pyright 1.1.167 from the command line and with the latest release of Pylance.

Code: 
```py
def fibonacci():
    a, b = 1, 1
    while True:
        yield a
        a, b = b, a + b

reveal_type(fibonacci)
```

Actual output:
```
() -> Generator[Literal[1], None, None]
```
Expected output:
```
() -> Generator[int, None, None]
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.168, which I just published. It will also be included in the next release of pylance.
